mithril.d.ts

```diff
@@ -596,7 +731,32 @@ declare module _mithril {
 		*
 		* @see m.component
 		*/
-		view(ctrl: T): MithrilVirtualElement<T>;
+		view(ctrl: T): MithrilVirtualElement;
+	}
+
+	/**
+	* This represents a parameterized MithrilComponent with up to six parameters,
+	* intended to be instantiated via m.component().
+	*
+	* @see m
+	* @see m.component
+	*/
+	interface ParameterizedMithrilComponent<TController extends MithrilController, T1, T2, T3, T4> {
+		/**
+		* The component's controller.
+		*
+		* @see m.component
+		*/
+		controller: (arg1: T1, arg2: T2, arg3: T3, arg4: T4) => TController |
+		  {new(arg1: T1, arg2: T2, arg3: T3, arg4: T4): TController};
```

Comment:
Don't use `new` for this part. Use the function call form for now. Currently, the controller is created with `.apply()`, and the prototype is set to the component itself.